### ISSUE TITLE
`no-unused-vars` warning when variable in module scope is named `globalThis`

### ISSUE THREAD

**Issue by 0f-0b** [denoland/deno_lint#1205]
### Lint Name

`no-unused-vars`

### Code Snippet

```js
const globalThis = 0;
console.log(globalThis);
```

### Expected Result

No warning.

### Actual Result

```shellsession
$ dlint run a.js
no-unused-vars

  × `globalThis` is never used
   ╭─[a.js:1:7]
 1 │ const globalThis = 0;
   ·       ─────┬────
   ·            ╰── If this is intentional, prefix it with an underscore like `_globalThis`
 2 │ console.log(globalThis);
   ╰────
  help: https://lint.deno.land/#no-unused-vars

Found 1 problem
```

### Version

```shellsession
$ dlint --version
dlint 0.52.2
```

Deno versions since denoland/deno@8f065a60e79e221a6ce7f6ce06c3022a85edb56a are affected.

**Comment by lucacasonato** [comment]
Upstream issue: https://github.com/swc-project/swc/issues/8317

**Comment by 0f-0b** [comment]
Fixed in 0.53.0.
